README.md

```diff
@@ -113,45 +158,55 @@ Signed Redemption Records are only accessible via a new option to the Fetch API:
+In order to achieve this, the browser generates the public/private keypair at redemption time, and includes the hash of the public key in the redemption request to the token issuer, which is included in the `ClientData` portion of the SRR. The keypair is then stored in new first-party storage only accessible via these APIs.
 
 An additional parameter to the Fetch API then allows the browser to include a signature over the SRR, request data, and additional request headers (specified using a new opt-in header), using the browser's private key associated with the SRR:
 
 
 ```
 fetch(<resource-url>, {
   ...
-  signedRedemptionRecord: <issuer>,
-  signRequestData: include | omit | headers-only,
+  trust-token: {
+    type: 'send-srr',
+    issuer: <issuer>,
+    refresh: false | true,
+    signRequestData: include | omit | headers-only,
+    includeTimestampHeader: false | true,
+    additionalSignedHeaders: <headers>
+  }
   headers: new Headers('Signed-Headers', '"sec-signed-redemption-record", "referer"')
   ... 
 });
 ```
 
 
-If `signRequestData` is `include`, then the browser will sign over the request data with the private key associated with the Signed Redemption Record. It will sign over the entire URL, POST body, public key, and any headers included in the `Signed-Headers` request header. If `signRequestData` is `headers-only`, we will only sign over those in `Signed-Headers`. Additionally we will limit the size of POST bodies that can be sent via this API when signed, to prevent having to buffer the entire body in memory. To do this, we can (aligning as close as possible to the model in the [Signed Exchanges](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#rfc.section.3.2) spec) create a canonical [CBOR](https://cbor.io) representation of the resource URL, public key, and headers:
+If `signRequestData` is `include`, then the browser will sign over the request data with the private key associated with the Signed Redemption Record. It will sign over the entire URL, POST body, public key, and any headers included in the `Signed-Headers` request header. If `signRequestData` is `headers-only`, we will only sign over those in `Signed-Headers`. If `includeTimestampHeader` is set, we will add a new header to the request `Sec-Time` with the current client time (which can optionally be added to `Signed-Headers`). Any headers included in `additionalSignedHeaders` will also be appended to `Signed-Headers`. Additionally we will limit the size of POST bodies that can be sent via this API when signed, to prevent having to buffer the entire body in memory. To do this, we can (aligning as close as possible to the model in the [Signed Exchanges](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#rfc.section.3.2) spec) create a canonical [CBOR](https://cbor.io) representation of the resource URL, public key, and headers:
 
 
 ```
 {
```

Comment:
optional: not sure how important it is but this CBOR is not canonically sorted:
https://tools.ietf.org/html/rfc7049#section-3.9

Reply:
As long as both sides can construct the same CBOR, this should be fine?

Reply:
Yeah I only mentioned this for clarity since we probably want to use the canonical sorting to avoid needing to specify it.